File: 1756-design-most-recently-used-queue/1756-design-most-recently-used-queue.java
class MRUQueue {

    List<Integer> list=new ArrayList<>();
    public MRUQueue(int n) {
        int i;
        for(i=1;i<=n;i++)
        {
            list.add(i);
        }
    }
    
    public int fetch(int k) {
        
        int temp=list.get(k-1);
        list.remove(k-1);
        list.add(temp);
        return temp;
    }
}

/**
 * Your MRUQueue object will be instantiated and called as such:
 * MRUQueue obj = new MRUQueue(n);
 * int param_1 = obj.fetch(k);
 */

File: 1756-design-most-recently-used-queue/1756-design-most-recently-used-queue.cpp
class MRUQueue {
public:
    list<int> li;
    list<int> :: iterator it;
    MRUQueue(int n) {
        int i;
        li.clear();
        for(i=1;i<=n;i++)
        {
            li.push_back(i);
        }
    }
    
    int fetch(int k) {
       int count=0;
        for(it=li.begin();it!=li.end();it++)
        {
            count++;
            if(k==count)
            {
                int val=*it;
                li.erase(it);
                li.push_back(val);
                return val;
                break;
            }
        }
        return -1;
    }
};

/**
 * Your MRUQueue object will be instantiated and called as such:
 * MRUQueue* obj = new MRUQueue(n);
 * int param_1 = obj->fetch(k);
 */